consensus/queryer.go

```diff
@@ -0,0 +1,82 @@
+package consensus
+
+import (
+	"context"
+
+	"github.com/ipfs/go-hamt-ipld"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/go-filecoin/abi"
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/state"
+	"github.com/filecoin-project/go-filecoin/types"
+	"github.com/filecoin-project/go-filecoin/vm"
+)
+
+// Abstracts over a store of blockchain state.
+type chainStateChainReader interface {
+	GetHead() types.TipSetKey
+	GetTipSetState(context.Context, types.TipSetKey) (state.Tree, error)
+	GetTipSet(types.TipSetKey) (types.TipSet, error)
+}
+
+// ChainState knows how to send read-only messages for querying actor state.
+type ChainState struct {
```

Comment:
I'm not 100% this is the right place for this (maybe it should be in its own package).  However this abstraction is sorely needed and has my vote.